### ISSUE TITLE
non constant name being incorrectly flagged as having invalied constant naming convention

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#124]
Originally reported by: **Walter Prins (BitBucket: [ByteJuggler](http://bitbucket.org/ByteJuggler), GitHub: @ByteJuggler?)**

---

I've been turning on informational/convention checking in PyLint (used under PyDev, but that's besides the point) and noticed PyLint complaining in many places about what it identified as constant naming violations, but which in fact was normal variables.  

I've isolated a testcase as below that demonstrates this issue.  

Can someone either confirm this as a bug or set me straight if not?

```
#!python

#test1.py
"""A docstring."""

def _patch_file(path, content):
    """A docstring."""
    return path, content

def _no_sandbox(function):
    """A docstring."""
    return function

_patch_file = _no_sandbox(_patch_file)
```

The (truncated) output is as follows:

```
walter-desktop[C:\Users\walterp\src\pylint\test\input]|61> !pylint test1.py
No config file found, using default configuration
************* Module input.test1
C: 12,0: Invalid name "_patch_file" for type constant (should match (([A-Z_][A-Z0-9_]*)|(__.*__))$)


Report
======
6 statements analysed.

Duplication
-----------

+-------------------------+------+---------+-----------+
|                         |now   |previous |difference |
+=========================+======+=========+===========+
|nb duplicated lines      |0     |0        |=          |
+-------------------------+------+---------+-----------+
|percent duplicated lines |0.000 |0.000    |=          |
+-------------------------+------+---------+-----------+
```

The version of pylint used is the latest available as of this writing -- I checked it out just prior to producing this test case.  pylint itself reports the version as follows:

```
walter-desktop[C:\Users\walterp\src\pylint\test\input]|64> !pylint --version
No config file found, using default configuration
pylint-script.py 0.25.2,
astng 0.24.3, common 0.59.1
Python 2.7.5 (default, May 15 2013, 22:43:36) [MSC v.1500 32 bit (Intel)]
```

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/124
